### PR TITLE
Allow empty arguments for arrayZip/arrayZipUnaligned

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):

- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow empty arguments for arrayZip/arrayZipUnaligned, as concat did in https://github.com/ClickHouse/ClickHouse/pull/65887. It is for spark compatiability in Gluten CH Backend.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit f9335a2fd5e4746a8b3fa9bb704640e1a4d3e61c with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69576/f9335a2fd5e4746a8b3fa9bb704640e1a4d3e61c/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/f62cb32c5c9f94e4d661b6a00c012564af5200f7/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/01045_array_zip.reference [hunk 1]
```diff
@@ -1,2 +1,3 @@
 [('a','d'),('b','e'),('c','f')]
 [('a','d','g'),('b','e','h'),('c','f','i')]
+[()]
```
Comment (alexkats) [inline tests/queries/0_stateless/01045_array_zip.reference:3]: It actually returns `[]` and that's why FastTest has failed, and it makes sense to me that there shouldn't be an empty tuple when there are no arrays at all

File: tests/queries/0_stateless/03230_array_zip_unaligned.reference [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [('a','d'),('b','e'),('c','f')]	Array(Tuple(Nullable(String), Nullable(String)))
 [('a','d','g'),('b','e','h'),('c','f','i')]
+[()]
```
Comment (alexkats) [inline tests/queries/0_stateless/03230_array_zip_unaligned.reference:3]: Same

**Review by alexkats — CHANGES_REQUESTED** [review]
In general lgtm

**Review by alexkats — APPROVED** [review]

**Comment by taiyang-li** [comment]
failed test in https://s3.amazonaws.com/clickhouse-test-reports/69576/71553022e0c36b7740445d1806f132613a394066/stateless_tests__release__old_analyzer__s3__databasereplicated__[1_2].html seems not related to this pr.

**Comment by alexkats** [comment]
Stateless tests (release, old analyzer, s3, DatabaseReplicated) [1/2]: #69624

**Pull request merged**